### PR TITLE
Accessibility fixes:

### PR DESCRIPTION
**Pull request by royeden**
Hello, what really neat library :smile:

I was checking it out for inspiration for a personal project (that will use a similar x/y component, but not for color picking) and came across some minor accessibility issues that I tried to fix right here:
* Keyboard color marker out of bounds: The keyboard movement allowed for the color marker to go out of bounds, this PR fixes that by applying the same contstrains as the pointer drag events.
* ~focus-visible: [`focus-visible`](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible#browser_compatibility) is currently supported on all browsers, this removes overhead of modifying classes via `classList` and it's simpler (it also seems to simplify the selectors a bit more).~
* ~Closing the picker with keyboard without compromising state: The `Esc` key would close the picker resetting it to it's previous state. This would clash on accessibility for keyboard only users, as it's the only way they have to close the dialog that pops up. (I left a comment just in case this should be reverted or you want to change the `revert` param on the `closePicker` function)~
* Keep focus on the color marker after a click: when clicking the color marker, I would expect the marker to have focus, but it wouldn't track focus for the element. This PR fixes that!

Also, I tested all this using a simple HTML file, it could maybe go inside an `examples/index.html` on the project:

```html
<!DOCTYPE html>
<!-- examples/index.html -->
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
  <link rel="stylesheet" href="/dist/coloris.css">
</head>
<body>
  <input type="text" data-coloris>
  <script href="/dist/coloris.js"></script>
</body>
</html>
```

### CODE REVIEW

**Comment by mdbassit** [comment]
Hello Roy, thank you for the kind words and for taking the time to debug this tool. I really appreciate it.

You are absolutely right about the color marker (that's what I called in the code anyway) going out of bounds when using the keyboard, thank you for the fix.

As for focus-visible, I'm aware that it is supported in all browsers, but it's still fairly new for most browsers, and some only implemented less than a year ago (Safari). I generally wait until a feature has been around for 4 years (arbitrary number) before I implement it and remove the old workarounds. I did a lot of work in Africa and Asia, and users in those regions tend to not update their browsers as often.

And lastly, there is actually a close button in the color picker that is not labeled but has an aria-label for people using a screen reader. But you are right that it is not useful for people who use keyboard navigation without a screen reader. I'll have to think of a way to make the close button obvious without compromising the sleek and simple design.

So in summary, I can only accept the out of bounds marker fix for now. Could I trouble you to resubmit a PR with only that fix?

Thanks,
M.

**Comment by royeden** [comment]
No problem, I've updated the code & the comment!

**Comment by melloware** [comment]
So to follow up how do we close the popup with keyboard only?   Or that needs to be figured out?

**Review by mdbassit — CHANGES_REQUESTED** [review]
Could you please remove the dist files from the commit?

**Comment by mdbassit** [comment]
> So to follow up how do we close the popup with keyboard only? Or that needs to be figured out?

Right now, the close button is not obvious unless you use a screen reader.

Screenshot 2023-01-11 at 10 40 23

Maybe I can add an optional more obvious close button similar to the optional clear button. Here is what the clear button looks like today:

Screenshot 2023-01-11 at 10 46 26

Any thoughts @melloware, @royeden?

**Comment by royeden** [comment]
Yes please, a visible close button would be great!

How would you distinguish between using that button and using `Escape`? Does `Escape` still reset state?

File: src/coloris.js [hunk 1]
```diff
@@ -579,6 +579,9 @@
     // Prevent scrolling while dragging the marker
     event.preventDefault();
     event.stopPropagation();
+
+    // After moving the marker, it should remain focused
+    colorMarker.focus()
```
[the remark this answers is not in the document]
Comment (royeden) [reply src/coloris.js:584]: Focus usually goes to the component you interacted with your mouse, this happens with any input, but I was thinking more about input with `type="range"`, so I was confused to find that pointer interactions didn't set focus on `color marker`.

In regards to showing the focus ring, all my ideas rely on redundancy and emulating `:focus-visible`, so I'm not absolutely sure which one to pick, leaving them here for future reference :

```js
// Option 1: After focusing the colorMarker 
colorMarker.addEventListener("focus", () => picker.classList.add('clr-keyboard-nav'))

// Option 2: In moveMarker after focusing the colorMarker:
function moveMarker(event) {
  /* rest of the code */
  // After moving the marker, it should remain focused
  colorMarker.focus()
  picker.classList.add('clr-keyboard-nav')
}

// Option 3: In moveMarkerOnKeydown:
function moveMarkerOnKeydown(offsetX, offsetY) {
  /* anywhere as order is not important for this action*/
  picker.classList.add('clr-keyboard-nav')
}
```

**Comment by royeden** [comment]
> Could you please remove the dist files from the commit?

Done!

**Comment by melloware** [comment]
@mdbassit Not sure if this helps but this DatePicker I use Has the buttons Clear and Close at the bottom on opposite sides...

image

**Review by mdbassit — APPROVED** [review]
I removed the call to `focus()' on the color marker

**Comment by mdbassit** [comment]
@royeden I just pushed a commit to make sure the focus rings are always visible when using the keyboard. This commit also focuses the marker when moved like you suggested.

**Comment by mdbassit** [comment]
> Yes please, a visible close button would be great!
> 
> How would you distinguish between using that button and using `Escape`? Does `Escape` still reset state?

`Escape` will still reset the state since the role of that key is to interrupt and/or reset. I'm working on adding an optional visible close button.

**Comment by melloware** [comment]
Nice work!

**Comment by mdbassit** [comment]
I just pushed a new release (v0.17.0) that adds an optional visible close button.

**Comment by melloware** [comment]
0.17.0 deployed to NPM as well!

**Comment by mdbassit** [comment]
@melloware I stupidly introduced a bug in v0.17.0 that I fixed in v0.17.1. Could I trouble you to update the NPM version to include the bugfix?

**Comment by melloware** [comment]
Done!

**Comment by mdbassit** [comment]
Thanks a lot!
